test/libyul/yulOptimizerTests/loopInvariantCodeMotion/no_move_state_recursive_function.yul

```diff
@@ -0,0 +1,26 @@
+{
+  function f() -> x { x := g() }
+  function g() -> x { x := g() }
+
+  let b := 1
+  for { let a := 1 } iszero(eq(a, 10)) { a := add(a, 1) } {
+    let t := extcodesize(f())
+    let q := sload(g())
+  }
+}
+// ----
+// step: loopInvariantCodeMotion
+//
+// {
+//     function f() -> x
+//     { x := g() }
+//     function g() -> x_1
+//     { x_1 := g() }
+//     let b := 1
+//     let a := 1
+//     for { } iszero(eq(a, 10)) { a := add(a, 1) }
+//     {
+//         let t := extcodesize(f())
```

Comment:
Why isn't it moved?

Reply:
Calling `f` causes infinite recursion and the for loop condition might be false.